launcher/ui/dialogs/MSALoginDialog.cpp

```diff
@@ -146,27 +146,32 @@ void MSALoginDialog::authorizeWithBrowser(const QUrl& url)
     m_url = url;
 }
 
-// https://stackoverflow.com/questions/21400254/how-to-draw-a-qr-code-with-qt-in-native-c-c
-void paintQR(QPainter& painter, const QSize sz, const QString& data, QColor fg)
+void paintQR(QPainter& painter, const QSize canvasSize, const QString& data, QColor fg)
 {
-    // NOTE: At this point you will use the API to get the encoding and format you want, instead of my hardcoded stuff:
-    qrcodegen::QrCode qr = qrcodegen::QrCode::encodeText(data.toUtf8().constData(), qrcodegen::QrCode::Ecc::LOW);
-    const int s = qr.getSize() > 0 ? qr.getSize() : 1;
-    const double w = sz.width();
-    const double h = sz.height();
-    const double aspect = w / h;
-    const double size = ((aspect > 1.0) ? h : w);
-    const double scale = size / (s + 2);
-    // NOTE: For performance reasons my implementation only draws the foreground parts in supplied color.
-    // It expects background to be prepared already (in white or whatever is preferred).
+    const auto* qr = QRcode_encodeString(data.toUtf8().constData(), 0, QRecLevel::QR_ECLEVEL_M, QRencodeMode::QR_MODE_8, 1);
+    if (!qr) {
+        qWarning() << "Unable to encode '" << data << "' as QR code";
```

Comment:
After intense scrutiny I found another issue - qt adds the spaces and quotes automatically
```suggestion
        qWarning() << "Unable to encode" << data << "as QR code";
```
(as if this code will ever run anyway)